glpi-project/glpi
# Enhance db:check command

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

- Move code to a dedicated and testable class;
- Check for indexes differences;
- Check for table options differences;
- Adapt checks to DB config, DB server and DB version;
- Add parameters to adapt checks depending on already ran migrations;
- Exit with an error code when differences are found;
- Permit do do a non strict check (do not take care or fields / index orders in table);
- Call the command after each migration on test suite;
- Add tests on most relevant cases.
